src/cartesian/Area.tsx

```diff
@@ -1,79 +1,90 @@
 /**
  * @fileOverview Area
  */
-import React, { PureComponent } from 'react';
-import PropTypes from 'prop-types';
+import React, { PureComponent, ReactElement } from 'react';
 import classNames from 'classnames';
+// @ts-ignore
 import Animate from 'react-smooth';
 import _ from 'lodash';
-import Curve from '../shape/Curve';
-import Dot from '../shape/Dot';
+import Curve, { CurveType, Point as CurvePoint } from '../shape/Curve';
+import Dot, { Props as DotProps } from '../shape/Dot';
 import Layer from '../container/Layer';
 import LabelList from '../component/LabelList';
-import { PRESENTATION_ATTRIBUTES, EVENT_ATTRIBUTES, LEGEND_TYPES, TOOLTIP_TYPES,
-  getPresentationAttributes, isSsr, filterEventAttributes } from '../util/ReactUtils';
+// @ts-ignore
+import { isSsr } from '../util/ReactUtils';
+// @ts-ignore
 import { isNumber, uniqueId, interpolateNumber } from '../util/DataUtils';
+// @ts-ignore
 import { getCateCoordinateOfLine, getValueByDataKey } from '../util/ChartUtils';
+import { Props as XAxisProps } from './XAxis';
+import { Props as YAxisProps } from './YAxis';
+import { D3Scale, LegendType, TooltipType, AnimationTiming, PresentationAttributes, filterProps, adaptEventHandlers, ChartOffset, Coordinate } from '../util/types';
+
+type AreaDot = ReactElement<SVGElement> | ((props: any) => SVGElement) | DotProps | boolean;
+interface AreaPointItem extends CurvePoint {
+  value?: number | number[];
+  payload?: any;
+}
 
 
-class Area extends PureComponent {
+interface InternalAreaProps {
+  xAxis?: Omit<XAxisProps, 'scale'> & { scale: D3Scale<string | number> };
+  yAxis?: Omit<YAxisProps, 'scale'> & { scale: D3Scale<string | number> };
```

Comment:
I found this pattern a lot in your code. How about creating `InternalXAxisProps` and `InternalYAxisProps`?